src/main/java/edu/ksu/canvas/attendance/submitter/CanvasAssignmentAssistant.java

```diff
@@ -111,15 +111,21 @@ public void editAssignmentInCanvas(Long courseId, AttendanceAssignment attendanc
     }
 
     public void deleteAssignmentInCanvas(Long canvasCourseId, OauthToken oauthToken) throws AttendanceAssignmentException{
-
         List<AttendanceSection> sections = attendanceSectionService.getSectionByCanvasCourseId(canvasCourseId);
         if(CollectionUtils.isEmpty(sections)) {
-            throw new AttendanceAssignmentException(AttendanceAssignmentException.Error.NON_EXISTENT_SECTION_ERROR);
+            LOG.warn("Sections empty");
         }
 
-        AttendanceAssignment assignment = assignmentRepository.findByAttendanceSection(sections.get(0));
+        AttendanceAssignment assignment = assignmentRepository.findByAttendanceSection(sections.get(0));;
+
+        Optional<Assignment> canvasAssignment = Optional.empty();
+        try{
+            canvasAssignment = canvasApiWrapperService.getSingleAssignment(canvasCourseId, oauthToken, assignment.getCanvasAssignmentId().toString());
+        } catch(IOException e){
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1166)